# Customers Segmentation using Keras

In [2]:
# General use libraries
import pandas as pd
import matplotlib.pyplot as plt 
import seaborn as sns
import numpy as np

In [6]:
# Importing dataset
df = pd.read_csv("data_customer_classification.csv")
df.head()

customer_id trans_date  tran_amount
0      CS5295  11-Feb-13           35
1      CS4768  15-Mar-15           39
2      CS2122  26-Feb-13           52
3      CS1217  16-Nov-11           99
4      CS1850  20-Nov-13           78

## Exploration
- Search for NA values and not normal observations
- Searching for relevant information about the data

In [7]:
print(len(df))

125000


In [12]:
df.nunique(axis=0)

customer_id    6889
trans_date     1401
tran_amount      96
dtype: int64

There are 6889 unique customers from 125,000 observations, it is possible to make combinations for each customer

In [13]:
#% of NAs for each column. No NA's in the dataset
nas = pd.DataFrame(df.isna().sum()/df.shape[0], columns = ['%NAs']).reset_index()
nas

index  %NAs
0  customer_id   0.0
1   trans_date   0.0
2  tran_amount   0.0

In [15]:
df.describe()

tran_amount
count  125000.000000
mean       64.991912
std        22.860006
min        10.000000
25%        47.000000
50%        65.000000
75%        83.000000
max       105.000000